charts/partials/templates/_proxy.tpl

```diff
@@ -39,7 +39,7 @@ env:
   value: 10000ms
 - name: LINKERD2_PROXY_OUTBOUND_CONNECT_KEEPALIVE
   value: 10000ms
-{{ if or (.Values.global.proxy.trace.collectorSvcAddr) (.Values.global.controlPlaneTracing) -}}
+{{ if or (.Values.global.proxy.trace.collectorSvcAddr) (and (hasPrefix "linkerd-" .Values.global.proxy.component) .Values.global.controlPlaneTracing) -}}
```

Comment:
My thinking is that if `collectorSvcAddr` is set for `controlPlaneTracing`, it could be then propogated and saved into `linkerd-config` which is used with application injections and making injections also have tracing fields set to the same.

Trying to find away to only have them set temporarily not stored.